### PR TITLE
SDK Update - Customer Service

### PR DESCRIPTION
**Pull request by imvanzen**
Added following methods:
- Create Customer
- Update Customer
- Fetch Customer
- Delete Customer

All documented, verified

### CODE REVIEW

File: src/main/java/pl/rspective/voucherify/client/api/VoucherifyApi.java [hunk 1]
```diff
@@ -140,4 +142,44 @@
...
+
+    /**
+     * Fetch a Customer using Customer ID
+     *
+     * @param customerId (required)
+     *
+     * @return customer with given ID
+     */
+    @GET("/customers/{id}")
+    Customer fetchCustomer(@Path("id") String customerId);
```
Comment (tpindel) [inline src/main/java/pl/rspective/voucherify/client/api/VoucherifyApi.java]: @imvanzen I know that in voucher we have `fetch` but let's change to name `getCustomer`. Thanks

File: src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+     * @return updated customer
+     */
+    public Customer updateCustomer(Customer customer) {
+        return api.updateCustomer(customer.getId(), customer);
+    }
+
+    /**
+     * Delete a customer.
+     *
+     * @param customer
```
Comment (LukaszWiktor) [inline src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java]: build failed
> error: @param name not found

File: src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+         * @param customer
+         *          customer to be created
+         *
+         */
+        public void createCustomer(Customer customer, VoucherifyCallback<Customer> callback) {
+            RxUtils.subscribe(executor, rx().createCustomer(customer), callback);
+        }
+
+        /**
+         * Create a customer.
```
Comment (LukaszWiktor) [inline src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java]: create or update?

File: src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+         *          customer to be created
+         *
+         */
+        public void updateCustomer(Customer customer, VoucherifyCallback<Customer> callback) {
+            RxUtils.subscribe(executor, rx().updateCustomer(customer), callback);
+        }
+
+
+        /**
+         * Create a customer.
```
Comment (LukaszWiktor) [inline src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java]: create or delete?

File: src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+                    return CustomerModule.this.updateCustomer(customer);
+                }
+            });
+        }
+
+        /**
+         * Delete a customer by customerId.
+         *
+         * @param customerId
+         *          customerId to be created
```
Comment (LukaszWiktor) [inline src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java]: to be created or deleted?

File: src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+            return RxUtils.defer(new RxUtils.DefFunc<Customer>() {
+                @Override
+                public Customer method() {
+                    return CustomerModule.this.createCustomer(customer);
+                }
+            });
+        }
+
+        /**
+         * Create a customer.
```
Comment (LukaszWiktor) [inline src/main/java/pl/rspective/voucherify/client/module/CustomerModule.java]: create?

**Pull request merged**